xdsl/builder.py

```diff
@@ -0,0 +1,131 @@
+from __future__ import annotations
+
+from typing import Callable, TypeAlias, overload
+
+from dataclasses import dataclass
+
+from xdsl.ir import OperationInvT, Attribute, Region, Block, BlockArgument
+from xdsl.dialects.builtin import FunctionType
+
+
+@dataclass
+class OpBuilder:
```

Comment:
I see the point, but I'm not 100% on board. Because the actual MLIR OpBuilder behaves quite different to this one, and if we move forward with this plan, the API of this class will change completely. I don't really like this, as it creates an unstable API on purpose, if you understand what I mean?

I really like this API, but it shares no methods with the MLIR `OpBuilder` class. That's why I'm raising this issue...

Reply:
Ah but it will once the Rewriter inherits from it :)

Reply:
Ah but it will once we add the insertion point